### ISSUE TITLE
Adding feature

### ISSUE THREAD

**Issue by GregSilverman** [dpgaspar/fab_addon_audit#1]
Hi,
I would like to modify the audit mechanism. The issue is that we are dealing with medical record data and need to keep a more robust audit trail. I am considering implementing something like this (although I would modify your audit_log class and have a single table for audits across all tables - instead of having multiple audit tables, as per this blog post - by utilizing a generalized structure with table_name/target, column_name, old_value, new_value, operation and record_key). See: http://blog.appliedinformaticsinc.com/implementing-hipaa-compliant-audit-trails-using-sqlalchemy-orm-queries/

Any pointers on how to do this would be most welcome. I will gladly contribute this back to your project.

**Comment by GregSilverman** [comment]
I'm also looking where I can modify the content of the message being passed to the AuditLog class. It appears that this is different between views that get updated. I'd like to make this more consistent, in particular, I would like it to be the primary key on the model bound to the view.

**Comment by GregSilverman** [comment]
I figured out how to grab the data AFTER the operation was done - by using the self.datamodel.get_values_item() method - but am wondering how to do this BEFORE the operation is done, so that I can log the value of the record before it changed. I played around with a new method pre_update(), but it looks like it is grabbing the data after the updates have been applied. Is there a method available that exposes the value of the "dirty" columns before the update?
